modules/@apostrophecms/piece-type/index.js

```diff
@@ -1059,11 +1059,12 @@ module.exports = {
   extendMethods(self) {
     return {
       getBrowserData(_super, req) {
+        const userBatchOperations = self.batchOperations.filter(batchOperation => self.apos.permission.can(req, 'edit', self.name, batchOperation.action));
```

Comment:
It would be better to tag the batch operations themselves with an optional `permission` property, which is what would be checked for here, if it is present. Generally it will be `edit`, or nothing (if you can view it you can do it), but that is an extensible pattern.

Reply:
Please factor this check to a method as it is likely to be extended often.